XenonCode.hpp

```diff
@@ -1552,15 +1556,21 @@ const int VERSION_PATCH = 0;
 						} else
 						
 					// function
-						if (words[0] == "function") {
-							if (words.size() > 1 && words[1] != Word::Funcname) {
+						if (words[0] == "function" || words[0] == "recursive") {
+							int offset = 0;
+							if (words[0] == "recursive" && words[1] != "function") {
```

Comment:
this would cause index out of range if the line only contains the word `recursive`
need to check for words.size()